### PR TITLE
Adds Development and Testing section

### PR DESCRIPTION
**Pull request by mustafaiman**
This section is mostly the same with the old one. The PR is mostly about paraphrasing and style

### CODE REVIEW

**Comment by mustafaiman** [comment]
@Serdaro @furkansenharputlu can you please review?

File: README.md [hunk 1]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
 ```
 
 
-# Development
+# Development and Testing
 
-## Building And Installing from Sources
+Hazelcast Node.js Client is developed using Typescript. If you want to help with bug fixes, develop new features or
```
Comment (Serdaro) [inline README.md]: Let's be consistent when stating "Node.js Client". I think, in most of the document, we use it as "Node.js client" ("c" is not capitalized).

File: README.md [hunk 2]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
...
-# Testing
+At this point you have all the runnable code in `lib` directory. You may create a link to this module so that your local
+applications can depend on your local copy of Hazelcast Node.js Client. In order to create a link, run:
+- `npm link`.
+This will create a global link to this module in your computer. Whenever you need to depend on this module from another
+local project, run:
+- `npm link hazelcast-client`
 
-## Testing Prerequisites
+If you are planning to contribute, please run style checker and fix the reported issues before sending a pull request.
```
Comment (Serdaro) [inline README.md]: ..., please run the style checker, as shown below, and fix the ...

File: README.md [hunk 3]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
 ```
 
 
-# Development
+# Development and Testing
 
-## Building And Installing from Sources
+Hazelcast Node.js client is developed using Typescript. If you want to help with bug fixes, develop new features or
```
Comment (furkansenharputlu) [inline README.md]: ... by using TypeScript.

File: README.md [hunk 4]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
...
 
-## Using Locally Installed Package
+Follow the below steps to build and install Hazelcast Node.js client from its source:
 
-- Change directory to the project directory.
-- Use a locally linked module: `npm link hazelcast-client`.
-- Tryout the example shown in [Using the Client](#using-the-client) section.
+1. Clone the GitHub repository (https://github.com/hazelcast/hazelcast-nodejs-client.git).
+2. Run `npm install` to automatically download and install all required modules under `node_modules' directory.
+3. Run  `npm run compile` to compile Typescript files to Javascript.
```
Comment (furkansenharputlu) [inline README.md]: ...to compile TypeScript files to JavaScript.

File: README.md [hunk 5]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
...
 
-# Testing
+At this point you have all the runnable code in `lib` directory. You may create a link to this module so that your local
```
Comment (furkansenharputlu) [inline README.md]: We may mention about generated `.d.ts` files.

File: README.md (hunk 3 above)
Comment (furkansenharputlu) [inline README.md]: Hazelcast Node.js Client..

File: README.md [hunk 6]
```diff
@@ -146,43 +146,41 @@ readData(dataInput) {
...
-
-- Clone the GitHub repository [https://github.com/hazelcast/hazelcast-nodejs-client.git](https://github.com/hazelcast/hazelcast-nodejs-client.git).
-- Install the dependencies using the command `npm install`.
-- Compile TypeScript using the command `npm run compile`.
-- Link the package locally using the command `npm link`.
-- Run static analysis tool using the command `npm run lint`
+## Building and Using Client From Sources
 
-## Using Locally Installed Package
+Follow the below steps to build and install Hazelcast Node.js client from its source:
```
Comment (furkansenharputlu) [inline README.md]: ...install Hazelcast Node.js Client..

Comment (mustafaiman) [reply README.md]: @Serdaro advised to be consistent and use `Hazelcast Node.js client` in one of the other PRs. I could not find his specific comment but that is what I remember. Maybe he can confirm.

Comment (furkansenharputlu) [reply README.md]: I was using as `Hazelcast Node.js Client`. @Serdaro Can we clarify?

Comment (Serdaro) [reply README.md]: @furkansenharputlu , most of the "client" in this Readme is expressed as "client" which is also a convention in out other public facing documents. So: "Hazelcast Node.js client"

**Comment by mustafaiman** [comment]
@Serdaro can you approve if you dont have other comments?

**Pull request merged**